src/mpid/ch4/src/ch4_globals.c

```diff
@@ -152,21 +152,27 @@ int MPIDI_check_for_failed_procs(void)
 #ifdef USE_PMIX_API
     MPIR_Assert(0);
 #elif defined(USE_PMI2_API)
-    len = PMI2_MAX_VALLEN;
     {
         int vallen = 0;
+        len = PMI2_MAX_VALLEN;
+        kvsname = MPL_malloc(len, MPL_MEM_OTHER);
+        MPIR_Assert(kvsname);
         pmi_errno =
             PMI2_KVS_Get(kvsname, PMI2_ID_NULL, "PMI_dead_processes", failed_procs_string,
                          len, &vallen);
         MPIR_ERR_CHKANDJUMP(pmi_errno, mpi_errno, MPI_ERR_OTHER, "**pmi_kvs_get");
+        MPL_free(kvsname);
     }
 #else
     pmi_errno = PMI_KVS_Get_value_length_max(&len);
     MPIR_ERR_CHKANDJUMP(pmi_errno, mpi_errno, MPI_ERR_OTHER, "**pmi_kvs_get_value_length_max");
+    kvsname = MPL_malloc(len, MPL_MEM_OTHER);
+    MPIR_Assert(kvsname);
     pmi_errno = PMI_KVS_Get_my_name(kvsname, len);
```

Comment:
So it seems that ch4 stores the kvsname value in the strangely named `MPIDI_CH4_Global.jobid`. We should use that and remove all the extra PMI calls here for now.